pkg/controller/cronjob/cronjob_controller.go

```diff
@@ -128,7 +128,15 @@ func (jm *CronJobController) syncAll() {
 	glog.V(4).Infof("Found %d groups", len(jobsBySj))
 
 	for _, sj := range sjs {
-		syncOne(&sj, jobsBySj[sj.UID], time.Now(), jm.jobControl, jm.sjControl, jm.podControl, jm.recorder)
+		tzTime, err := getCurrentTimeInZone(&sj)
+		if err != nil {
+			// We validate that the CronJob time zone is valid.
+			// However, it is possible that the list of time zones could change at runtime which would cause an error when getting the time.
+			jm.recorder.Eventf(&sj, v1.EventTypeWarning, "InvalidTimeZone", "Attempted to run a job with an invalid time zone: %v", sj.Spec.TimeZone)
```

Comment:
Can you include the error in the event message?

Reply:
Please, address this comment alongside the rebase. Things like this tend to fall through to obscurity if there is no accompanying issue and this change is small enough that I don't think it warrants an issue.